pkg/kv/util/weak_owner.go

```diff
@@ -244,10 +245,44 @@ func (w *WeakOwner) startOwningKey(ctx context.Context, owner string, prefixedKe
 	}
 }
 
-// Own blocks until it gets weak ownership of key for owner.  Ownership
-// will be refreshed at resolution interval.  It returns a function to stop
-// owning key.
+// releaseIf releases prefixedKey if it has the owner.
+func (w *WeakOwner) releaseIf(ctx context.Context, owner string, prefixedKey []byte) error {
+	log := w.Log.WithContext(ctx).WithFields(logging.Fields{
+		"prefixed_key": string(prefixedKey),
+		"owner":        owner,
+	})
+	ownership := WeakOwnership{}
+	predicate, err := kv.GetMsg(ctx, w.Store, weakOwnerPartition, prefixedKey, &ownership)
+	if err != nil {
+		return fmt.Errorf("get ownership message %s to release it from %s: %w", string(prefixedKey), owner, err)
+	}
+	log = log.WithField("new_owner", ownership.Owner)
+
+	if ownership.Owner != owner {
+		log.Info("Lost ownership race before trying to release")
+		return nil
+	}
+	ownership.Owner = fmt.Sprintf("[released] %s", ownership.Owner)
```

Comment:
I don't think we need to change this. Let's keep the Owner as is, you have the 0 `Expires` field and `Comment` if you need to add info. It's best to keep the same format if we can.